Rubberduck.CodeAnalysis/Inspections/Concrete/VariableNotUsedInspection.cs

```diff
@@ -39,15 +39,24 @@ namespace Rubberduck.Inspections.Concrete
     /// </example>
     public sealed class VariableNotUsedInspection : InspectionBase
     {
+        /// <summary>
+        /// Inspection results for variables that are never referenced.
+        /// </summary>
+        /// <returns></returns>
         public VariableNotUsedInspection(RubberduckParserState state) : base(state) { }
 
+        /// <summary>
+        /// VariableNotUsedInspection override of InspectionBase.DoGetInspectionResults()
+        /// </summary>
+        /// <returns>Enumerable IInspectionResults</returns>
         protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
         {
             var declarations = State.DeclarationFinder.UserDeclarations(DeclarationType.Variable)
                 .Where(declaration =>
                     !declaration.IsWithEvents
                     && !declaration.IsIgnoringInspectionResultFor(AnnotationName)
-                    && !declaration.References.Any());
+                    && (!declaration.References.Any() 
+                        || declaration.References.All(rf => rf.IsAssignment)));
```

Comment:
The second condition alone should suffice. If there are no elements, `All` will always return `true`.

Reply:
Wow...that result for `All` on an empty collection is really not intuitive to me...but you're right.  Done.